File: DSA/Cpp/src/dsa/lib/lists/arraylist/arraylist.cpp
#include "arraylist.h"

namespace dsa::lib::lists {

ArrayList::ArrayList() {
    int ary[DEFAULT_LENGTH];
    this->lst = ary;
}
ArrayList::ArrayList(const int ary[]) {
    this->lst = ary;
}
ArrayList::~ArrayList() {
    delete[] lst;
}
length_t ArrayList::length() {
//    ToDo
    return 0;
}

}  // namespace dsa

File: DSA/Java/src/dsa/container/lists/ArrayList.java
package dsa.container.lists;

import dsa.lib.lists.List;

public class ArrayList implements List {
    private static final int DEFAULT_LENGTH = 10;
    private final int[] lst;

    public ArrayList() {
        this.lst = new int[DEFAULT_LENGTH];
    }

    @Override
    public int size() {
        return 0;
    }
}
